backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/diagrams/NodeMappingStyleProvider.java

```diff
@@ -86,13 +90,32 @@ private RectangularNodeStyle createRectangularNodeStyle(VariableManager variable
 
         Result result = this.interpreter.evaluateExpression(variableManager.getVariables(), squareDescription.getBorderSizeComputationExpression());
         int borderSize = result.asInt().getAsInt();
+        Integer width = squareDescription.getWidth() * SIZE_FACTOR;
+        Integer height = squareDescription.getHeight() * SIZE_FACTOR;
+
+        // If the initial width and/or height have not been set by the specifier, we interpret the size computation
+        // expression to set the width and/or height
+        if (width == 0 || height == 0) {
+            result = this.interpreter.evaluateExpression(variableManager.getVariables(), squareDescription.getSizeComputationExpression());
+            int size = result.asInt().getAsInt() * SIZE_FACTOR;
+            if (size > 0) {
+                if (width == 0) {
+                    width = size;
+                }
+                if (height == 0) {
+                    height = size;
+                }
+            }
+        }
 
         // @formatter:off
         return RectangularNodeStyle.newRectangularNodeStyle()
                 .color(color)
                 .borderColor(borderColor)
                 .borderSize(borderSize)
                 .borderStyle(borderStyle)
+                .width(width)
+                .height(height)
```

Comment:
It seems that you can pretty much rewrite the whole commit on the size of elements computed from the VSM. This information has no place here.